### DISCUSSION TITLE
How to compute duration from datetime

### DISCUSSION THREAD

**Discussion by mcanouil** [typst/typst#2758] [Q&A] [answered]
According to the documentation, it is possible, but I can't seem to figure out how.
Note that the documentation only says "You can also get a duration by subtracting two [datetimes](https://typst.app/docs/reference/foundations/datetime/)." which is not really helpful.

Assume I have the following two dates and wants to extract to number of days, how can I compute the duration without triggering "cannot joint content"?

```typst
let x = datetime(
  year: 2012,
  month: 8,
  day: 3,
)

let y = datetime.today()
```

Any help would be very much appreciated.

**Comment by LaurenzV** [comment, marked as answer]
This should work:

```typst
#let x = datetime(
  year: 2012,
  month: 8,
  day: 3,
)

#let y = datetime.today()
#let duration = y - x

Days between the dates: #duration.days()
```

**Reply by mcanouil** [reply]
Hum, okay I need to better understand types of my objects especially when defining function and using them in another function when creating a template.

Thanks for confirming the way to compute the duration, it helped me focused on the rest.
